src/pudl/transform/ferc1.py

```diff
@@ -1661,10 +1803,60 @@ def _multiplicative_error_correction(tofix, mask, minval, maxval, mults):
     return fixed
 
 
+def rename_table(
+    raw_table: pd.DataFrame, table_name: str, source: Literal["xbrl", "dbf"]
+) -> pd.DataFrame:
+    """Rename FERC1 table based on table name and source."""
+    return raw_table.rename(columns=COLUMN_RENAME[table_name][source])
+
+
+def merge_instant_and_duration_tables(
+    duration: pd.DataFrame, instant: pd.DataFrame, merge_on: list[str]
+) -> pd.DataFrame:
+    """Merge the XBRL instand and duration tables.
+
+    FERC1 XBRL instant period signifies that it is true as of the reported date,
+    while a duration fact is true for the specified time period. The ``date``
+    column for an instant fact is effectively equivalent to the ``end_date``
+    column of a duration fact.
+
+    Args:
+        duration: XBRL duration table.
+        instant: XBRL instant table.
+        merge_on: columns to merge on besides the date columns.
+
+    Returns:
+        one unified table that is a combination of the duration and instant
+        input tables. Any overlapping columns (besides the ``merge_on`` columns)
+        will be consolidated by filling the duration columns with the instant
+        columns.
```

Comment:
@zschira  does this treatment of the overlapping columns in the instant v duration tables seem reasonable? The only overlapping column from the steam table is `filing_name`. I don't even think we are going to keep the filing name column. For this table the data overlapped perfectly except for nulls in from the instant table.

After thinking about this, I almost lean toward either keeping both w/ suffixes through the transform step in case they are useful or just dropping all overlapping columns.

Reply:
There actually should never be any overlapping columns except for `filing_name`, `entity_id`, and `Axis` columns. The rest of the columns should only be present in the `duration` table, or the `instant` table, not both. In fact, if we wanted to we could programmatically merge on `entity_id` and any `Axis` columns, rather than passing specifying `merge_on` explicitly.

Reply:
great! this is good news. i can chuck some of this overlapping code.